### PR TITLE
Make payload limit configurable.

### PR DESCRIPTION
**Pull request by markusthoemmes**
The maximum payload allowed to flow through the system is hardcoded today. Several pieces (kafka for instance) need to be adjusted to allow a larger payload.

**Note:** After travis passes happily, the default can be reset to 1 MB to not change anything here.

### CODE REVIEW

File: common/scala/src/main/resources/application.conf [hunk 1]
```diff
@@ -70,9 +70,10 @@ whisk {
                 retention-ms    = 3600000
             }
             invoker {
-                segment-bytes   =  536870912
-                retention-bytes = 1073741824
-                retention-ms    = 172800000
+                segment-bytes     =  536870912
+                retention-bytes   = 1073741824
+                retention-ms      =  172800000
+                max-message-bytes = ${whisk.activation.payload.max}
```
Comment (rabbah) [inline common/scala/src/main/resources/application.conf]: Why is this needed?

File: common/scala/src/main/resources/application.conf [hunk 2]
```diff
@@ -88,6 +89,12 @@ whisk {
         stride = ${?CONTROLLER_INSTANCES}
     }
 
+    activation {
+        payload {
+            max = 5242880 // 5 m not possible because cross-referenced to kafka configurations
```
Comment (rabbah) [inline common/scala/src/main/resources/application.conf]: Vs just this.

File: common/scala/src/main/scala/whisk/connector/kafka/KafkaProducerConnector.scala [hunk 3]
```diff
@@ -87,6 +87,9 @@ class KafkaProducerConnector(kafkahosts: String,
     val props = new Properties
     props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkahosts)
     props.put(ProducerConfig.ACKS_CONFIG, 1.toString)
+    // Allow messages with up to 5 MiB size
+    // The target topic must support this message size as well because it's larger as the default
+    props.put(ProducerConfig.MAX_REQUEST_SIZE_CONFIG, 5242880.toString)
```
Comment (rabbah) [inline common/scala/src/main/scala/whisk/connector/kafka/KafkaProducerConnector.scala]: Should the 5M also be a setting here from pure config ?

File: common/scala/src/main/resources/application.conf (hunk 1 above)
Comment (markusthoemmes) [reply common/scala/src/main/resources/application.conf]: Wanted to seperate kafka and general config, so the value is carried over by configuration not by hardcoding.

I also changed the topic's configurations to be generalized as in: You can provide all possible Kafka keys here and they are transformed to the respective Kafka string.

File: common/scala/src/main/scala/whisk/connector/kafka/KafkaProducerConnector.scala (hunk 3 above)
Comment (markusthoemmes) [reply common/scala/src/main/scala/whisk/connector/kafka/KafkaProducerConnector.scala]: Presumably, yes. Need to thing how to cleverly thread that through 👍

**Comment by markusthoemmes** [comment]
PG2 2692 🔵
